### PR TITLE
Remove Swift 5.0 and 5.1 support, and add CI config

### PR DESCRIPTION
**Pull request by Davidde94**
NIOTS only had CI configs for Swift 5.0, so I've added a few more to bring our coverage up to match other members of the NIO family. Additionally update `Package.swift` to drop support for 5.0 and 5.1.

### CODE REVIEW

**Comment by Davidde94** [comment]
Hey @yim-lee this one is a bit more involved. Could you please:

- [ ] Delete the old 5.0 job
- [ ] Make new jobs for 5.2, 5.3, 5.4, and main

Johannes added main in a separate PR #114

**Comment by yim-lee** [comment]
@swift-nio-bot test this please

**Comment by yim-lee** [comment]
@Davidde94 CI changes done.

I added 5.2, 5.3, and 5.4 jobs to required list.

Unrelated to this, but I have created a [soundness job](https://ci.swiftserver.group/job/swift-nio-transport-services-soundness-prb/). It should be added to the required list as well but currently it's failing because we need to update `scripts/sanity.sh` and `docker-compose.yaml` first.

**Comment by Davidde94** [comment]
@yim-lee thanks a lot, I'll fix it now

File: docker/docker-compose.1804.53.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+version: "3"
+
+services:
+
+  runtime-setup:
+    image: swift-nio-transport-services:18.04-5.3
+    build:
+      args:
+        ubuntu_version: "xenial"
```
Comment (Lukasa) [inline docker/docker-compose.1804.53.yaml:9]:
```suggestion
        ubuntu_version: "bionic"
```

File: docker/docker-compose.2004.54.yaml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+version: "3"
+
+services:
+
+  runtime-setup:
+    image: swift-nio-transport-services:18.04-5.4
```
Comment (Lukasa) [inline docker/docker-compose.2004.54.yaml:6]:
```suggestion
    image: swift-nio-transport-services:20.04-5.4
```

File: docker/docker-compose.2004.54.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    build:
+      args:
+        ubuntu_version: "focal"
+        swift_version: "5.4"
+
+  test:
+    image: swift-nio-transport-services:18.04-5.4
```
Comment (Lukasa) [inline docker/docker-compose.2004.54.yaml:13]:
```suggestion
    image: swift-nio-transport-services:20.04-5.4
```

File: docker/docker-compose.2004.54.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  shell:
+    image: swift-nio-transport-services:18.04-5.4
```
Comment (Lukasa) [inline docker/docker-compose.2004.54.yaml:16]:
```suggestion
    image: swift-nio-transport-services:20.04-5.4
```

**Review by Lukasa — CHANGES_REQUESTED** [review]
While we're here, I know it's not your fault @Davidde94 but the nightly dockerfile is just _nonsense_. Can you rewrite it to match the ones here instead of just being a copy from NIO?

**Comment by Davidde94** [comment]
Yep no problem

**Review by Lukasa — APPROVED** [review]
LGTM.

**Pull request merged**